vllm/config/compilation.py

```diff
@@ -523,9 +558,16 @@ def init_backend(self, vllm_config: "VllmConfig") -> Union[str, Callable]:
             if self.backend in torch_backends:
                 return self.backend
             return resolve_obj_by_qualname(self.backend)
+        if self.level == CompilationLevel.PIECEWISE:
+            if self.backend == "":
```

Comment:
You can move this below the assert and remove the `if`. Also I think by this point we should have modified the `backend` to something other than `""`